terraform/dns.tf

```diff
@@ -405,29 +400,34 @@ locals {
       value    = "2a01:4f9:c012:8178::"
     },
 
-    # Mailserver configuration for `mail-test.nixos.org`
+    # Mailserver configuration for `nixos.org`
+    # TODO: remove the 2 MX records for improvmx below in favor of this once
+    # we're ready to switch to the new mailserver:
+    # https://github.com/NixOS/infra/issues/485
+    # {
+    #   hostname = "nixos.org"
+    #   type     = "MX"
+    #   value    = "umbriel.nixos.org"
+    # },
     {
-      hostname = "mail-test.nixos.org"
-      type     = "MX"
-      value    = "umbriel.nixos.org"
-    },
-    {
-      hostname = "mail-test.nixos.org"
+      hostname = "nixos.org"
       type     = "TXT"
-      value    = "v=spf1 mx ~all"
+      # TODO: simplify to just a `mx` rule once umbriel is our one and only
+      # mailserver:
+      # https://github.com/NixOS/infra/issues/485
+      # value = "v=spf1 mx ~all"
+      value = "v=spf1 mx include:spf.improvmx.com include:_mailcust.gandi.net ip4:37.27.20.162 ip6:2a01:4f9:c011:8fb5::1 ~all"
```

Comment:
A couple notes:

1. I *think* we could get rid of the hardcoded IPs and replace them with `a:umbriel.nixos.org`. I didn't originally do that because the `a:` sounded to me like ipv4 only, but after doing [some reading](https://www.reddit.com/r/DMARC/comments/1884e8w/a_record_in_include_how_does_it_resolve/) I think that's not the case.
2. What's the deal with this `_mailcust.gandi.net`? Is it safe to remove?

Reply:
I asked @zimbatm about gandi. I have the vague memory that this might be no longer used. but unclear.

Reply:
Maybe leave a comment in pluto that we also need to update the MX record if the IP of the machine changes.

Reply:
Marking this resolved.

1. I've switched this to `a:umbriel.nixos.org`. I'll confirm it works with both ipv4 and ipv6 sending once deployed.
2. I've rebased this on top of https://github.com/NixOS/infra/commit/ad70ad1e9ec8397f33146cc6e8d192b8c2ac47a8, which removes the `gandi.net` stuff.